examples/connect.rs

```diff
@@ -124,82 +124,40 @@ fn main() {
+                        println!("");
+                        println!(
+                            "Copy this info and share it with your connecting partner. Then paste \
+                          their info below."
+                        );
+                        read_line().and_then(move |line| {
+                            let their_pub_info: PubConnectionInfo<PeerId> = unwrap!(
                         serde_json::from_str(&line)
                     );
 
-                    service
-                    .connect(our_priv_info, their_pub_info)
-                    .map_err(|e| panic!("error connecting to peer: {}", e))
-                    .and_then(move |peer| {
-                        let (peer_sink, peer_stream) = peer.split();
-
-                        println!("You are now connected! say hello :)");
-                        let writer = {
-                            future::loop_fn(peer_sink, |peer_sink| {
-                                read_line()
-                                .and_then(|line| {
-                                    let line = line.into_bytes();
-                                    peer_sink
-                                    .send((0, line))
-                                    .map(Loop::Continue)
-                                    .map_err(|e| panic!("error sending message to peer: {}", e))
-                                })
-                            })
-                        };
-                        let reader = {
-                            peer_stream
-                            .map_err(|e| panic!("error receiving message from peer: {}", e))
-                            .for_each(|line| {
-                                let line = match String::from_utf8(line) {
-                                    Ok(line) => line,
-                                    Err(..) => String::from("<peer sent invalid utf8>"),
-                                };
-                                println!("{}", line);
-                                Ok(())
-                            })
-                            .map(|()| {
-                                println!("peer disconnected");
-                            })
-                        };
-
-                        writer
-                        .select2(reader)
-                        .map(|either| match either {
-                            Either::A((v, _next)) => void::unreachable(v),
-                            Either::B(((), _next)) => drop(service),
-                        })
-                        .map_err(|either| match either {
-                            Either::A((v, _next)) => v,
-                            Either::B((v, _next)) => v,
+                            service
+                                .connect(our_priv_info, their_pub_info)
+                                .map_err(|e| panic!("error connecting to peer: {}", e))
+                                .and_then(move |peer| have_a_conversation(service, peer))
```

Comment:
That's cool - really more readable, but I would still like to keep a minimal `connect` example and what if this one was a `chat` example instead?
Then we would have examples from least to most complex:
* connect - only connects two peers
* exchange_data - exchanges one message each
* chats - let's you send as many messages interactively as you want